model/task_lifecycle.go

```diff
@@ -220,10 +221,21 @@ func TryResetTask(taskId, user, origin string, detail *apimodels.TaskEndDetail)
 			return errors.Wrap(err, "Error marking task as ended")
 		}
 	}
+	if t.IsPartOfSingleHostTaskGroup() {
+		if err = t.SetResetWhenFinished(); err != nil {
+			return errors.Wrapf(err, "error setting resetWhenFinished for task")
 
-	if err = resetTask(t.Id, user); err != nil {
-		return err
+		}
+		// want to restart all tasks in the task group, if it's finished
+		if err = checkResetTaskGroup(t); err != nil {
+			return errors.Wrapf(err, "error resetting task group for task '%v'", t.Id)
+		}
+	} else {
+		if err = resetTask(t.Id, user); err != nil {
```

Comment:
it seems inconsistent that resetTask takes an ID but checkResetTaskGroup takes a task object.

Reply:
The main reason I have checkResetTaskGroup take a task is to be consistent with checkResetDisplayTask. That being said, I also think checkResetTaskGroup needs more than an ID (for example, we couldn't callI IsPartOfSingleHostTaskGroup without it).